src/compiler/types.ts

```diff
@@ -6238,6 +6238,7 @@ export interface NodeLinks {
     potentialReflectCollisions?: Node[];
     potentialUnusedRenamedBindingElementsInTypes?: BindingElement[];
     externalHelpersModule?: Symbol;     // Resolved symbol for the external helpers module
+    instantiationExpressionTypes?: Map<number, Type>; // Cache of instantiation expression types for the node

```

Comment:
Is there any reason you cache by IDs instead of by Type?

Reply:
Historic convention and habit, mostly. I can swap it if you'd like it.